### PR TITLE
INS-2921: fixed leak in ImmutableOrder and fixed OnPulse tests behaviour

### PR DESCRIPTION
**Pull request by sexyhamster**
**- What I did**
- fixed goroutine test leak
- fixed TestExecutionBroker_AddFreshRequestWithOnPulse logic

### CODE REVIEW

File: logicrunner/logicrunner_unit_test.go [hunk 1]
```diff
@@ -138,6 +139,7 @@ func (suite *LogicRunnerTestSuite) SetupLogicRunner() {
 }
 
 func (suite *LogicRunnerTestSuite) AfterTest(suiteName, testName string) {
+
```
Comment (bigbes) [inline logicrunner/logicrunner_unit_test.go]: nitpicking: excess newline was introduced

File: logicrunner/logicrunner_unit_test.go [hunk 2]
```diff
@@ -428,6 +430,7 @@ func TestLogicRunner_OnPulse_Order(t *testing.T) {
 
 func (suite *LogicRunnerTestSuite) TestImmutableOrder() {
 	syncT := &testutils.SyncT{T: suite.T()}
+	defer leaktest.Check(syncT)()
```
[the remark this answers is not in the document]
Comment (bigbes) [reply logicrunner/logicrunner_unit_test.go]: Also, I don't know if check `suite.True(wait(finishedCount, broker, 3))` is still needed.

[the remark this answers is not in the document]
Comment (sexyhamster) [reply logicrunner/logicrunner_unit_test.go]: leaktest.Check awaits up to 5 seconds.
there should not be race conditions

the check is still neeeded, to make sure that everything is processed

[the remark this answers is not in the document]
Comment (sexyhamster) [reply logicrunner/logicrunner_unit_test.go]: well, actually you are right, check could be removed

File: logicrunner/executionbroker.go [hunk 3]
```diff
@@ -577,6 +577,10 @@ func (q *ExecutionBroker) isClosed() bool {
 	}
 }
 
+func (q *ExecutionBroker) Close() {
```
[the remark this answers is not in the document]
Comment (sexyhamster) [reply logicrunner/executionbroker.go]: yep, will move it to lowerCase and will check isClosed

File: logicrunner/logicrunner_unit_test.go (hunk 1 above)
Comment (sexyhamster) [reply logicrunner/logicrunner_unit_test.go]: fixed

File: logicrunner/logicrunner_unit_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sexyhamster) [reply logicrunner/logicrunner_unit_test.go]: after 5 seconds everything should be 100% stopped

File: logicrunner/executionbroker.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sexyhamster) [reply logicrunner/executionbroker.go]: I've added isClosed check, but this function is for tests only, and is not thread safe

File: logicrunner/executionbroker_test.go [hunk 4]
```diff
@@ -548,6 +548,10 @@ func TestExecutionBroker_HasMoreRequestsWithOnPulse(t *testing.T) {
 			checks: func(ctx context.Context, t *testing.T, broker *ExecutionBroker, msgs []payload.Payload) {
 				require.Len(t, msgs, 1)
 
+				wait(func(...interface{}) bool {
```
Comment (bigbes) [inline logicrunner/executionbroker_test.go]: add "assert.True" here, pls

Comment (sexyhamster) [reply logicrunner/executionbroker_test.go]: i removed this block

Comment (bigbes) [reply logicrunner/executionbroker_test.go]: nvmnd

**Pull request merged**